resources/js/features/facility-edit/FacilityCreateForm.tsx

```diff
@@ -0,0 +1,35 @@
+import {createMutation} from "@tanstack/solid-query";
+import {useLangFunc} from "components/utils";
+import {Admin, System} from "data-access/memo-api";
+import {VoidComponent} from "solid-js";
+import toast from "solid-toast";
+import {FacilityForm, FacilityFormOutput} from "./FacilityForm";
+
+interface Props {
+  onSuccess?: () => void;
+  onCancel?: () => void;
+}
+
+export const FacilityCreateForm: VoidComponent<Props> = (props) => {
+  const t = useLangFunc();
+  const facilityMutation = createMutation(() => ({
+    mutationFn: Admin.createFacility,
+    meta: {isFormSubmit: true},
+  }));
+
+  async function createFacility(values: FacilityFormOutput) {
+    await facilityMutation.mutateAsync({
+      name: values.name,
+      url: values.url,
+    });
+    Admin.useInvalidator().facilities();
```

Comment:
No więc jest taka sprawa że podobno pewnych rzeczy nie należy robić w handlerach, a jedną z takich rzeczy jest pobieranie różnego rodzaju kontekstów. Dlatego ten adminInvalidate myślę że powinien być na zewnątrz utworzony a tu tylko zawołany. Ale nie mam twardych danych że to nie są tylko przesądy. Natomiast wiem że coś podobnego mi kiedyś nie zadziałało i Ryszard mówił coś tam o hookach czego nie zrozumiałem, ale znaczyło tyle że czegoś nie mogę w handlerze.

Reply:
To by się chyba zgadzało z tym co piszą o "event callback" tutaj: https://docs.solidjs.com/references/api-reference/component-apis/createContext#default-value